app/controllers/workflows_controller.rb

```diff
@@ -1,15 +1,137 @@
 # frozen_string_literal: true
 
 class WorkflowsController < Sellers::BaseController
-  before_action :set_body_id_as_app
+  before_action :set_workflow, only: %i[edit update destroy]
+  before_action :authorize_workflow, only: %i[edit update destroy]
+
+  layout "inertia"
+
+  FLASH_CHANGES_SAVED = "Changes saved!"
+  FLASH_WORKFLOW_PUBLISHED = "Workflow published!"
+  FLASH_WORKFLOW_UNPUBLISHED = "Unpublished!"
+  FLASH_WORKFLOW_DELETED = "Workflow deleted!"
+
+  inertia_share do
+    parent_shared_data = RenderingExtension.custom_context(view_context).merge(
+      current_user: current_user_props(current_user, impersonated_user),
+      authenticity_token: form_authenticity_token,
+      flash: inertia_flash_props,
+      title: @title
+    )
+
+    if @workflow
+      workflow_presenter = WorkflowPresenter.new(seller: current_seller, workflow: @workflow)
+      workflow_props = {
+        workflow: -> { workflow_presenter.workflow_props },
+        context: -> { workflow_presenter.workflow_form_context_props }
+      }
+      parent_shared_data.merge(workflow_props)
+    else
+      parent_shared_data
+    end
+  end
 
   def index
     authorize Workflow
     create_user_event("workflows_view")
+
+    workflows_presenter = WorkflowsPresenter.new(seller: current_seller)
+    render inertia: "Workflows/Index", props: workflows_presenter.workflows_props
+  end
+
+  def new
+    authorize Workflow
+    workflow_presenter = WorkflowPresenter.new(seller: current_seller)
+    render inertia: "Workflows/New", props: {
+      context: -> { workflow_presenter.workflow_form_context_props }
+    }
+  end
+
+  def create
+    authorize Workflow
+
+    success, errors = save_workflow
+    if success
+      redirect_to workflow_emails_path(@workflow.external_id), notice: FLASH_CHANGES_SAVED, status: :see_other
+    else
+      redirect_to new_workflow_path, inertia: { errors: errors }, status: :see_other
+    end
+  end
+
+  def edit
+    render inertia: "Workflows/Edit"
+  end
+
+  def update
+    success, errors = save_workflow
+    if success
+      notice_message = case workflow_params[:save_action_name]
+      when "save_and_publish"
+        FLASH_WORKFLOW_PUBLISHED
+      when "save_and_unpublish"
+        FLASH_WORKFLOW_UNPUBLISHED
```

Comment:
Why are these action names here? Shouldn't they always be passed to `workflows#update` now?